### PR TITLE
CLI for the Platform API!

### PR DESCRIPTION
**Pull request by brimoor**
This PR adds a command-line interface for the Platform API!

It is available through the `voxel51` executable, which is now automatically installed as a package script when this library is installed.

**EXAMPLE:** List my last 10 uploaded data
```
$ voxel51 data --list 10
id                                    name                                                   size bytes  type       upload date               expiration date
------------------------------------  ---------------------------------------------------  ------------  ---------  ------------------------  ------------------------
1b728eb5-e313-46dd-81e6-290d5dcaf064  GGT9_hospital.mp4                                        25304591  video/mp4  2019-11-09T17:11:33.000Z  2019-11-16T17:11:31.000Z
9fac4985-1cc6-4354-bc41-6512e868dd91  GGT9_Sick Song Nursery Rhymes Daddy songs .mp4           31548843  video/mp4  2019-11-09T17:11:27.000Z  2019-11-16T17:11:26.000Z
5b56e2ab-4af3-42cb-8740-29af9e8bb14e  GGT9_Empathy_ The Human Connection to Patient C ...      29825071  video/mp4  2019-11-09T17:11:25.000Z  2019-11-16T17:11:24.000Z
871536cb-0289-442f-bdce-1dd69de8bc6e  GGT9-sickness-bridesmaids_2011_movie_clip_dress ...      24081308  video/mp4  2019-11-09T17:11:24.000Z  2019-11-16T17:11:23.000Z
ab72097a-b058-4975-a215-db69b4bb8598  GGT9-sickness-TheCarrie.mp4                              53782339  video/mp4  2019-11-09T17:11:24.000Z  2019-11-16T17:11:23.000Z
fac0ca76-5f8c-4a57-ad12-da50796fdaa0  GGT9-hospital_series_terror_attack_victims_0pjU ...     218793638  video/mp4  2019-11-09T17:11:18.000Z  2019-11-16T17:11:17.000Z
36502d6e-ce87-4cae-bfbc-0348034b74f8  GGT9-hospital.mp4                                        64605842  video/mp4  2019-11-09T17:11:16.000Z  2019-11-16T17:11:15.000Z
072fc4d3-b709-46c6-8f09-0b787a683e10  GGT9-House M.D..mp4                                      30632156  video/mp4  2019-11-09T17:11:15.000Z  2019-11-16T17:11:15.000Z
f9800c06-ddb2-40fb-8732-2eb59044ba91  GGT9-drunk_and_drunker_untold_stories_of_the_er ...       9848221  video/mp4  2019-11-09T17:11:15.000Z  2019-11-16T17:11:15.000Z
2daff43b-2697-48d6-a49b-1da2959c68e0  GGT7_vehicle_accident_highway.mp4                        10769671  video/mp4  2019-11-09T17:11:09.000Z  2019-11-16T17:11:08.000Z
```

**EXAMPLE:** List my last 10 uploaded jobs 
```
$ voxel51 jobs --list 10
id                                    name                                                 state     archived    upload date               expiration date
------------------------------------  ---------------------------------------------------  --------  ----------  ------------------------  ------------------------
0b074fb6-551d-41a0-a9f8-1eb604af754c  gcs-threat-103-GGT9_hospital                         COMPLETE  False       2019-11-09T17:11:36.000Z  2019-11-12T17:11:37.000Z
7d2f12a9-e718-4aaf-9277-26b624836138  gcs-threat-103-GGT9_Sick Song Nursery Rhymes Da ...  COMPLETE  False       2019-11-09T17:11:32.000Z  2019-11-12T17:11:32.000Z
e144ef7c-eebd-4c77-b6be-205cca3cc7e9  gcs-threat-103-GGT9_Empathy_ The Human Connecti ...  COMPLETE  False       2019-11-09T17:11:31.000Z  2019-11-12T17:11:31.000Z
127ab31d-5b1e-462f-a2dc-9fce1cba5dac  gcs-threat-103-GGT9-sickness-TheCarrie               COMPLETE  False       2019-11-09T17:11:27.000Z  2019-11-12T17:11:27.000Z
58473c98-4f79-4c67-95c9-ef7797e1433b  gcs-threat-103-GGT9-sickness-bridesmaids_2011_m ...  COMPLETE  False       2019-11-09T17:11:27.000Z  2019-11-12T17:11:28.000Z
f4bb7275-94db-4c8c-8b38-5c56af87b011  gcs-threat-103-GGT9-hospital                         COMPLETE  False       2019-11-09T17:11:23.000Z  2019-11-12T17:11:24.000Z
112817b6-7b17-4681-826c-56fb5fc953ac  gcs-threat-103-GGT9-hospital_series_terror_atta ...  COMPLETE  False       2019-11-09T17:11:21.000Z  2019-11-12T17:11:22.000Z
0ac9733e-e747-4c6f-865a-f24f0857c18e  gcs-threat-103-GGT9-drunk_and_drunker_untold_st ...  COMPLETE  False       2019-11-09T17:11:20.000Z  2019-11-12T17:11:21.000Z
3e75837c-9fe0-4c31-8a8a-32522ec7b702  gcs-threat-103-GGT9-House M.D.                       COMPLETE  False       2019-11-09T17:11:18.000Z  2019-11-12T17:11:19.000Z
3e62597d-2b4e-49c8-9d65-b0b7dbee4fd5  gcs-threat-103-GGT7_top-5 Massive landslide Cau ...  COMPLETE  False       2019-11-09T17:11:15.000Z  2019-11-12T17:11:16.000Z
```

Overall help:
```
$ voxel51 --help
usage: voxel51 [-h] [-v] {auth,data,jobs,analytics} ...

Voxel51 Platform API command-line interface.

optional arguments:
  -h, --help            show this help message and exit
  -v, --version         show version info

available commands:
  {auth,data,jobs,analytics}
    auth                Tools for configuring API tokens.
    data                Tools for working with data.
    jobs                Tools for working with jobs.
    analytics           Tools for working with analytics.
```

Authentication-related documentation:
```
 $ voxel51 auth --help
usage: voxel51 auth [-h] {show,activate,clean} ...

Tools for configuring API tokens.

optional arguments:
  -h, --help            show this help message and exit

available commands:
  {show,activate,clean}
    show                Show info about the active API token.
    activate            Activate an API token.
    clean               Deletes the active API token, if any.
```

Data-related documentation:
```
$ voxel51 data --help
usage: voxel51 data [-h] {list,info,upload,download,delete} ...

Tools for working with data.

optional arguments:
  -h, --help            show this help message and exit

available commands:
  {list,info,upload,download,delete}
    list                List data uploaded to the platform.
    info                Get information about data uploaded to the platform.
    upload              Upload data to the platform.
    download            Download data from the platform.
    delete              Delete data from the platform.
```

Jobs-related documentation:
```
$ voxel51 jobs --help
usage: voxel51 jobs [-h]
                    {list,info,upload,start,archive,unarchive,request,status,log,download,kill,delete}
                    ...

Tools for working with jobs.

optional arguments:
  -h, --help            show this help message and exit

available commands:
  {list,info,upload,start,archive,unarchive,request,status,log,download,kill,delete}
    list                List jobs on the platform.
    info                Get information about jobs on the platform.
    upload              Upload job requests to the platform.
    start               Start jobs on the platform.
    archive             Archive jobs on the platform.
    unarchive           Unarchive jobs on the platform.
    request             Download job requests.
    status              Download job statuses.
    log                 Download job logfiles.
    download            Download job outputs.
    kill                Kill jobs on the platform.
    delete              Delete jobs on the platform.
```

Analytics-related documentation:
```
$ voxel51 analytics --help
usage: voxel51 analytics [-h] {list,docs,upload,delete} ...

Tools for working with analytics.

optional arguments:
  -h, --help            show this help message and exit

available commands:
  {list,docs,upload,delete}
    list                List analytics on the platform.
    docs                Get documentation about analytics.
    upload              Upload analytics to the platform.
    delete              Delete analytics from the platform.
```

### CODE REVIEW

File: voxel51/users/auth.py [hunk 1]
```diff
@@ -53,45 +53,81 @@ def deactivate_token():
...
+    token_path = os.environ.get(TOKEN_ENVIRON_VAR, None)
+    if token_path is not None:
+        if not os.path.isfile(token_path):
+            raise TokenError(
+                "No API token found at '%s=%s'" %
+                (TOKEN_ENVIRON_VAR, token_path))
+    elif os.path.isfile(TOKEN_PATH):
+        token_path = TOKEN_PATH
+    else:
+        raise TokenError("No API token found")
```
Comment (iantimmis) [inline voxel51/users/auth.py:82]: May be more user-friendly to say something like: `No API token for Voxel51 platform was found at /path/to/token. If you do not have a token, you can download one from https://console.voxel51.com/account/tokens`

Comment (iantimmis) [reply voxel51/users/auth.py]: (I think there are a few places in this PR that could use an update like this)

Comment (brimoor) [reply voxel51/users/auth.py:82]: That already happens. Look at the definition of `TokenError`

Comment (iantimmis) [reply voxel51/users/auth.py:82]: Oh, cool. Nevermind then!

**Review by iantimmis — APPROVED** [review]
LGTM

**Pull request merged**